cvmfs/ssl.cc

```diff
@@ -0,0 +1,106 @@
+/**
+ * This file is part of the CernVM File System.
+ */
+
+#ifndef CVMFS_SSL_H_
+#define CVMFS_SSL_H_
+
+#include "ssl.h"
+
+#include <cstdlib>
+#include <string>
+#include <vector>
+
+#include "duplex_curl.h"
+#include "platform.h"
+#include "util/posix.h"
+#include "util/string.h"
+
+#ifdef CVMFS_NAMESPACE_GUARD
+namespace CVMFS_NAMESPACE_GUARD {
+#endif
+
+bool HasCertificates(std::string directory) {
+  if (!DirectoryExists(directory)) return false;
+
+  DIR *dirp = opendir(directory.c_str());
+  if (!dirp) return false;
+
+  platform_dirent64 *dirent;
+  while ((dirent = platform_readdir(dirp))) {
+    if ((dirent->d_type == DT_REG) || (dirent->d_type == DT_UNKNOWN)) {
```

Comment:
Unfortunately, we cannot rely on `d_type`. Some file systems don't fill it.  One has to make an explicit `stat` call given `dirent->d_name` to keep the code portable.